Framework/Kernel/inc/MantidKernel/DataService.h

```diff
@@ -19,6 +19,15 @@
 
 namespace Mantid {
 namespace Kernel {
+
+/// Flag for whether to sort items before returning
+enum class DataServiceSort { Sorted, Unsorted };
+/**
+ * Flag for whether to include hidden items when returning,
+ * Auto queries the class to determine this behavior
+ */
+enum class DataServiceHidden { Auto, Include, Exclude };
```

Comment:
@FedeMPouzols David chatted to me about this and usually I would agree with you. What we didn't like much was that the templated type would need to be included when calling the enum e.g.  `DataService<Workspace_sptr>::Sorted`, so this seemed like a reasonable compromise. What do you think?